robot/impl/resource_manager_test.go

```diff
@@ -746,6 +746,19 @@ func TestManagerNewComponent(t *testing.T) {
 	test.That(t, err.Error(), test.ShouldContainSubstring, "circular dependency")
 	test.That(t, err.Error(), test.ShouldContainSubstring, "arm3")
 	test.That(t, err.Error(), test.ShouldContainSubstring, "board3")
+
+	// add a new config with the same name and it should error
+	diff.Added.Components = append(diff.Added.Components, resource.Config{
```

Comment:
Could we add some more testing for adding non-unique services, remotes, processes and modules?

Reply:
added some tests now!